src/components/common/MediaItem.tsx

```diff
@@ -0,0 +1,326 @@
+      '&:hover': {
+        transform: 'translate(-50%, -50%) scale(1.1)',
+      },
+    }),
+    []
+  );
+
+  const processingFallbackStyles: SxProps<Theme> = useMemo(
+    () => ({
+      alignItems: 'center',
+      aspectRatio: 4 / 3,
+      backgroundColor: theme.palette.TwClrBaseGray100,
+      borderColor: theme.palette.TwClrBg,
+      borderStyle: 'solid',
+      borderWidth: '4px',
+      boxSizing: 'content-box',
+      display: 'flex',
+      flexDirection: 'column',
+      justifyContent: 'center',
+      textAlign: 'center',
+      width: '100%',
+    }),
+    [theme]
+  );
+
+  const onClickDownload = useCallback(
+    (event?: React.MouseEvent<HTMLButtonElement, MouseEvent> | undefined) => {
+      event?.stopPropagation();
+
+      if (onDownload) {
+        onDownload(mediaFile.fileId);
+      } else if (downloadUrl) {
+        const link = document.createElement('a');
+        link.href = downloadUrl;
+        link.download = mediaFile.fileName;
+        document.body.appendChild(link);
+        link.click();
+        document.body.removeChild(link);
+      }
+    },
+    [downloadUrl, mediaFile.fileId, mediaFile.fileName, onDownload]
+  );
+
+  const onClickExpand = useCallback(
+    (event?: React.MouseEvent<HTMLButtonElement, MouseEvent> | React.MouseEvent<HTMLDivElement, MouseEvent>) => {
+      event?.stopPropagation();
+
+      // For videos, open lightbox and fetch stream
+      if (mediaFile.type === 'Video' && observationId && plotId) {
+        setLightboxOpen(true);
+        void getObservationMediaStream({
+          observationId,
+          plotId,
+          fileId: mediaFile.fileId,
+        });
+      } else {
+        // For photos, use the parent's expand handler
+        onExpand(mediaFile.fileId);
+      }
+    },
+    [mediaFile.fileId, mediaFile.type, observationId, plotId, getObservationMediaStream, onExpand]
+  );
+
+  const handleCloseLightbox = useCallback(() => {
+    setLightboxOpen(false);
+    setMediaStream(undefined);
+  }, []);
+
+  const handleImageError = useCallback(() => {
+    // only check for 412 error if this is a video
+    if (mediaFile.type === 'Video' && observationId && plotId) {
+      // Try to fetch the media stream to check if video is processing (412 error)
+      void getObservationMediaStream({
+        observationId,
+        plotId,
+        fileId: mediaFile.fileId,
+      });
+    }
+    if (onImageError) {
+      onImageError();
+    }
+  }, [getObservationMediaStream, mediaFile.fileId, mediaFile.type, observationId, onImageError, plotId]);
+
+  useEffect(() => {
+    if (mediaStreamError) {
+      // Check if the error is a 412 (Precondition Failed) - video is still processing
+      const error = mediaStreamError as { status?: number };
+      if (error.status === 412) {
+        setIsPreconditionFailedError(true);
+
+        // retry after one minute
+        const timeoutId = setTimeout(() => {
+          setIsPreconditionFailedError(false);
+        }, 60000);
```

Comment:
Should the error message in this case indicate that the user has to wait 60s before trying again? Or am I reading the code wrong?

Reply:
This is how it works in the activity log. It shows a message that the video is processing (line 252) and it retries each minute, when the video is uploaded the preview images appears and the message disappears